src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -132,7 +148,21 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                     return;
                 }
 
-                bindingContext.Result = ModelBindingResult.Success(model);
+                if (_isBindingRequired && model == null)
```

Comment:
Don't add a new property on the `InputFormatterResult`. It wouldn't help unless we forced changes in all of the input _serializers_ and they're external. E.g. the formatters themselves don't check for whitespace.

We really have only two choices. I prefer (2) but want to hear a third opinion. @rynowak ??
1. Use the formatter returning `null`. Feels a bit like our previous mixups with `[Required]` semantics.
2. Update line 108 of `InputFormatter` which is the closest thing we have to `IsModelSet`.